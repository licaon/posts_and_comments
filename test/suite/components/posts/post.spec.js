import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Post from 'src/components/posts/post';

describe('Post component', () => {
  const createComponent = (props) => {
    const wrapper = shallow(
      <Post
        onDelete={() => {}}
        onToggleComments={() => {}}
        {...props}
      />);
    return toJson(wrapper);
  };
  it('should render component correctly without comments', (done) => {
    const component = createComponent({
      post: {
        id: 1,
        title: 'title',
        body: 'body',
      },
    });

    expect(component).toMatchSnapshot();
    done();
  });
  it('should render component correctly with comments', (done) => {
    const component = createComponent({
      post: {
        id: 1,
        title: 'title',
        body: 'body',
      },
      showComments: true,
      onValueChange: () => {},
    });

    expect(component).toMatchSnapshot();
    done();
  });
});
