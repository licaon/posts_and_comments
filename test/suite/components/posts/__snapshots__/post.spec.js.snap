// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render component correctly with comments 1`] = `
<Card
  className="post"
  expanderIconChildren="keyboard_arrow_down"
  expanderIconClassName="material-icons"
  expanderTooltipPosition="left"
>
  <CardTitle
    title="title"
  />
  <CardText
    component="section"
  >
    body
  </CardText>
  <CardActions>
    <div
      className="action-buttons"
    >
      <ButtonTooltipedInked
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}
        label="Hide Comments"
        onClick={[Function]}
        raised={true}
      />
      <ButtonTooltipedInked
        className="delete-button"
        icon={true}
        iconClassName="fa fa-trash"
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}
        onClick={[Function]}
        secondary={true}
      />
    </div>
  </CardActions>
  <Connect(Comments)
    postId={1}
  />
</Card>
`;

exports[`Post component should render component correctly without comments 1`] = `
<Card
  className="post"
  expanderIconChildren="keyboard_arrow_down"
  expanderIconClassName="material-icons"
  expanderTooltipPosition="left"
>
  <CardTitle
    title="title"
  />
  <CardText
    component="section"
  >
    body
  </CardText>
  <CardActions>
    <div
      className="action-buttons"
    >
      <ButtonTooltipedInked
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}
        label="Show Comments"
        onClick={[Function]}
        raised={true}
      />
      <ButtonTooltipedInked
        className="delete-button"
        icon={true}
        iconClassName="fa fa-trash"
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}
        onClick={[Function]}
        secondary={true}
      />
    </div>
  </CardActions>
</Card>
`;
